Modules/Scripted/DICOMLib/DICOMUtils.py

```diff
@@ -0,0 +1,96 @@
+import os
+import vtk, qt, ctk, slicer
+import logging
+
+#########################################################
+#
+#
+comment = """
+
+  DICOMUtils is a collection of static-like DICOM
+  utility functions facilitating convenient scripted
+  use of the DICOM database
+
+"""
+#
+######################################################### 
+
+#------------------------------------------------------------------------------
+# Get patient UID by patient name for easy loading of a patient
+def patientUIDByPatientName(name):
+  if not hasattr(slicer,'dicomDatabase'):
+    logging.error('DICOM database cannot be accessed!')
+    return None
+  patients = slicer.dicomDatabase.patients()
+  for patientUID in patients:
+    currentName = slicer.dicomDatabase.nameForPatient(patientUID)
+    if currentName == name:
+      return patientUID
+  return None
+
+#------------------------------------------------------------------------------
+# Load patient by patient UID from DICOM database
+def loadPatientByUID(patientUID):
+  if not hasattr(slicer,'dicomDatabase') or not hasattr(slicer.modules,'dicom'):
+    logging.error('DICOM module or database cannot be accessed!')
+    return
+  dicomWidget = slicer.modules.dicom.widgetRepresentation().self()
+
+  # Select all series in selected patient
+  studies = slicer.dicomDatabase.studiesForPatient(patientUID)
+  series = [slicer.dicomDatabase.seriesForStudy(study) for study in studies]
+  seriesUIDs = [uid for uidList in series for uid in uidList]
+  dicomWidget.detailsPopup.offerLoadables(seriesUIDs, 'SeriesUIDList')
+  dicomWidget.detailsPopup.examineForLoading()
+
+  # Load selected data
+  dicomWidget.detailsPopup.loadCheckedLoadables()
+
+#------------------------------------------------------------------------------
+# Load patient by patient name from DICOM database
+def loadPatientByName(patientName):
+  loadPatientByUID(patientUIDByPatientName(patientName))
+
+#------------------------------------------------------------------------------
+# Open specified DICOM database
+def openDatabase(databaseDir):
+  if not hasattr(slicer,'dicomDatabase') or not hasattr(slicer.modules,'dicom'):
+    logging.error('DICOM module or database cannot be accessed!')
+    return
+  if not os.access(databaseDir, os.F_OK):
+    logging.error('Specified database directory ' + repr(databaseDir) + ' cannot be found!')
+    return
+  dicomWidget = slicer.modules.dicom.widgetRepresentation().self()
+  dicomWidget.onDatabaseDirectoryChanged(databaseDir)
+  if not slicer.dicomDatabase.isOpen:
+    logging.error('Unable to open DICOM database ' + databaseDir)
+
+#------------------------------------------------------------------------------
+# Open temporary DICOM database, return current database directory
+def openTemporaryDatabase(folderName=None):
```

Comment:
I would suggest to great a context manager. That would allow to write code like this:
```
with openTemporaryDatabase() as d:
    d.doSomething()
```

The database would then be closed when exiting the context.

Example here: https://github.com/Slicer/Slicer/blob/7cfc9d6fc9ec2480810582750908ac05922f3a27/Base/Python/slicer/util.py#L491-L500